ccmlib/common.py

```diff
@@ -551,19 +551,15 @@ def check_socket_available(itf):
 
 
 def check_socket_listening(itf, timeout=60):
-    end = time.time() + timeout
-    while time.time() <= end:
-        try:
-            sock = socket.socket()
-            sock.connect(itf)
-            sock.close()
-            return True
-        except socket.error:
-            # Try again in another 200ms
-            time.sleep(.2)
-            continue
+    def _check_socket_listening() -> bool:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as _socket:
```

Comment:
Nitpick: why add the AF_INET, SOCK_STREAM parameters, if it works without them?
What if in 10 years AF_INET6 becomes the default, and this code no longer works?

Reply:
Yes, you're right, but it's a rate case.
The old logic was wrong because `socket.socket()` fails, the socket not close...

Reply:
socket.socket() will not fail, it's the connect() that can fail. And this isn't related to my quesition why pass these parameters to socket() if the defaults were good enough. But never mind, it's not important enough to fuss about.

Reply:
I mean to the `connect` method that can fail and because of that the socket can be open

Reply:
Ok, but this is not related to my comment which was about the parameters to socket(), not to connect(). But never mind, it's fine with or without these parameters. I just thought it's better not to change it without reason.